contents/data_vis_aog.md

```diff
@@ -0,0 +1,231 @@
+# Data Visualization with AlgebraOfGraphics.jl {#sec:aog}
+
+> **_NOTE:_**
+> Don't forget to check [`AlgebraOfGraphics.jl` documentation](https://aog.makie.org).
+
+`AlgebraOfGraphics.jl` is a higher-level plotting package that uses `Makie.jl` under the hood.
+It is geared towards data visualization workflows with support for `DataFrame`s.
+`AlgebraOfGraphics.jl` abstracts away some common patterns in data visualization using an interface that was based on the "grammar of graphics" approach that is used by R's `ggplot2` package.
+
+The **grammar of graphics** is a framework that follows a layered approach to construct visualizations in a structure manner.
+There are four main types of layers:
+
+- **data** layer
+- **mapping** layer
+- **visual transformation** layer
+- **statistical transformation** layer
+
+`AlgebraOfGraphics.jl` allows you to construct all of these types of layers with functions that returns a `Layer` object,
+in which all of the information necessary will be encoded.
+You can then perform **two operations on layers**:
+
+- **multiplication with `*`**: this fuses two or more layers into a _single_ layer
+- **addition with `+`**: this superimposes two or more layers into a vector of `Layer`s
+
+Finally, as the name suggests, `AlgebraOfGraphics.jl` is more than a grammar but also an algebra for `Layer` objects.
+And, as such, it defines two algebraic properties.
+Let `a`, `b` and `c` be `Layer` objects:
+
+- **associative property**: `(a * b) * c = a * (b * c)`
+- **distributive property**: `a  * (b + c) = (a * b) + (a + b)`
+
+To get started with `AlgebraOfGraphics.jl`,
+you'll need to load it along with a desired `Makie.jl` backend (Chapter -@sec:DataVisualizationMakie):
+
+```julia
+using AlgebraOfGraphics
+using CairoMakie
+```
+
+## Layers {#sec:aog_layers}
+
+The first layer we'll cover will be the **data layer**.
+You can create data layers with the `data` function:
+
+```jl
+sco("""
+data_layer = data(grades_2020())
+""")
+```
+
+As you can see, `data` takes any `DataFrame` and returns a `Layer` type.
```

Comment:
maybe is worth mentioning that not only DataFrames are acceptable. Have you seen the latest examples here, https://beautiful.makie.org/dev/examples/generated/2d/aog/scatterlinesAoG/, maybe something is worth transferring here.

Reply:
Please check if [6df1648](https://github.com/JuliaDataScience/JuliaDataScience/pull/289/commits/6df164833c7e4554560c06bdb4988e77f96c9aa4) solves this.